apps/ex_wire/lib/ex_wire/sync.ex

```diff
@@ -281,23 +284,26 @@ defmodule ExWire.Sync do
           Blocktree.t(),
           Blocktree.t(),
           Trie.t(),
-          BlockQueue.t(),
-          integer()
-        ) :: {integer(), Trie.t()}
+          BlockQueue.t()
+        ) :: Trie.t()
   defp maybe_request_new_block_or_save(
          block_tree,
          next_block_tree,
          trie,
-         block_queue,
-         last_requested_block
+         block_queue
```

Comment:
I try to avoid nested statements. What if you pattern match on empty queue or matching block tree?